### ISSUE TITLE
OOM training on videos, works perfect on images

### ISSUE THREAD

**Issue by frosty3907** [kohya-ss/musubi-tuner#317]
RTX 3090 w 24GB VRAM, can train off an image dataset of 100's of images in about 5 minutes with great results.
But as soon as I try and use video, even a single 81 frame clip I am getting OOM CUDA crashes after the first sample step.
Here's my train command:
accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 E:\Musubi\musubi\musubi-tuner\hv_train_network.py --dit E:\Musubi\musubi\musubi-tuner\models\mp_rank_00_model_states.pt --dataset_config E:\Musubi\musubi\Datasets\CURRENT\training.toml --sdpa --mixed_precision bf16 --fp8_base --optimizer_type adamw8bit --learning_rate 1e-3 --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers --network_module=networks.lora --network_dim=32 --timestep_sampling sigmoid --discrete_flow_shift 2.0 --max_train_epochs 999 --save_every_n_epochs=1 --seed 42 --output_dir "E:\Musubi\Output Models" --output_name XORFUK1 --vae E:/Musubi/musubi/musubi-tuner/models/pytorch_model.pt --vae_chunk_size 32 --vae_spatial_tile_sample_min_size 128 --text_encoder1 E:\ComfyUI_windows_portable\ComfyUI\models\text_encoders\llava_llama3_fp16.safetensors --text_encoder2 E:\ComfyUI_windows_portable\ComfyUI\models\clip\clip_l.safetensors --sample_prompts E:\Musubi\prompts.txt --sample_every_n_steps 1 --sample_at_first

And here's the training.toml:
[general]

resolution = [512, 384]
caption_extension = ".txt"
batch_size = 1
enable_bucket = true
bucket_no_upscale = false

[[datasets]]

video_directory = "E:\\Musubi\\musubi\\Datasets\\CURRENT"
cache_directory = "E:\\Musubi\\musubi\\Datasets\\CURRENT\\cache"
frame_extraction = "full"
max_frames = 81

e: running windows

**Comment by frosty3907** [comment]
Ah, I thought I read someone was using 81 frames with 24gb vram. Sounds like I need to aim way lower. 
I had trouble using the alternative frame extraction modes previously, but will give it another try.

**Comment by dorpxam** [comment]
The "chunk" extraction method is not more than the full length split into parts. I currently train a concept LoRA of 15 shorts length 768x512 videos. Using my settings, the cache become:

```
001-001_00000-037_0768x0512_hv.safetensors
001-001_00037-037_0768x0512_hv.safetensors
001-001_hv_te.safetensors
001-002_00000-037_0768x0512_hv.safetensors
001-002_00037-037_0768x0512_hv.safetensors
001-002_hv_te.safetensors
002-001_00000-037_0768x0512_hv.safetensors
002-001_00037-037_0768x0512_hv.safetensors
002-001_hv_te.safetensors
002-002_00000-037_0768x0512_hv.safetensors
002-002_hv_te.safetensors
003-001_00000-037_0768x0512_hv.safetensors
003-001_hv_te.safetensors
003-002_00000-037_0768x0512_hv.safetensors
003-002_hv_te.safetensors
004-001_00000-037_0768x0512_hv.safetensors
004-001_00037-037_0768x0512_hv.safetensors
004-001_hv_te.safetensors
004-002_00000-037_0768x0512_hv.safetensors
004-002_hv_te.safetensors
005-001_00000-037_0768x0512_hv.safetensors
005-001_00037-037_0768x0512_hv.safetensors
005-001_hv_te.safetensors
005-002_00000-037_0768x0512_hv.safetensors
005-002_00037-037_0768x0512_hv.safetensors
005-002_00074-037_0768x0512_hv.safetensors
005-002_hv_te.safetensors
006-001_00000-037_0768x0512_hv.safetensors
006-001_00037-037_0768x0512_hv.safetensors
006-001_hv_te.safetensors
007-001_00000-037_0768x0512_hv.safetensors
007-001_00037-037_0768x0512_hv.safetensors
007-001_hv_te.safetensors
008-001_00000-037_0768x0512_hv.safetensors
008-001_00037-037_0768x0512_hv.safetensors
008-001_00074-037_0768x0512_hv.safetensors
008-001_hv_te.safetensors
009-001_00000-037_0768x0512_hv.safetensors
009-001_00037-037_0768x0512_hv.safetensors
009-001_hv_te.safetensors
009-002_00000-037_0768x0512_hv.safetensors
009-002_00037-037_0768x0512_hv.safetensors
009-002_hv_te.safetensors
```

As you can see, the videos are splitted by 37 frames chunks and truncated to the most close part. Ofcourse there's only one text encoder cache by video.

**Comment by dorpxam** [comment]
OK. So the problem is probably not at this level. 

Did you have already set the optimization tricks ? Say, `--fp8_base`, `--split_attn` or maybe `--blocks_to_swap` ?

What is your cache latent settings ? The VAE chunk size for example ?

**Comment by frosty3907** [comment]
ok trying now, I note you have --flash_attn and no --sdpa as in mine -could that be relevant?

**Comment by frosty3907** [comment]
Ok so to clarify, my dataset at 512*384 would oom until I used fp8_llm, then it used 19gb. 
When I change my dataset to 320*240 vram usage was only 16gb, and when I remove fp8_llm, it is STILL only 16gb... Is there anyway I can get it to use a bit more of the available vram?

**Comment by Meliauo** [comment]
> Ok so to clarify, my dataset at 512_384 would oom until I used fp8_llm, then it used 19gb. When I change my dataset to 320_240 vram usage was only 16gb, and when I remove fp8_llm, it is STILL only 16gb... Is there anyway I can get it to use a bit more of the available vram?

--blocks_to_swap set to 0

**Comment by kohya-ss** [comment]
> Ok so to clarify, my dataset at 512_384 would oom until I used fp8_llm, then it used 19gb._

It's a bit strange that you get OOM at 512x384 without --fp8_llm. Could you please share the stack trace of that?

**Comment by frosty3907** [comment]
> > Ok so to clarify, my dataset at 512_384 would oom until I used fp8_llm, then it used 19gb._
> 
> It's a bit strange that you get OOM at 512x384 without --fp8_llm. Could you please share the stack trace of that?

Can't provide stack trace because the video adapter locks up, I can maybe get one mouse action in every five minutes.
Still testing various files and resolutions.

**Comment by frosty3907** [comment]
> > Is there anyway I can get it to use a bit more of the available vram?
> 
> I can't help you for that. About your OOM problem, maybe you can check and share more details of your VRAM usage during the training process. If you run with a 24GB GPU, I don't think the system is the problem, there's probably 3 or 4 GB used by something unidentified now.

No I check task manager pretty religiously, less than 1gb in use before training, not running anything else at the same time.

**Comment by kohya-ss** [comment]
> Can't provide stack trace because the video adapter locks up, I can maybe get one mouse action in every five minutes.

I got it. In my test of hv_train_network.py, once training started, the memory usage did not change whether fp8_llm was used or not. Please let me know if you notice anything.

**Comment by frosty3907** [comment]
> About your OOM problem, maybe you can check and share more details of your VRAM usage during the training process. If you run with a 24GB GPU, I don't think the system is the problem, there's probably 3 or 4 GB used by something unidentified now.

One thing I've noticed is that it'll use about 14GB VRAM to do the initial sample prompt render, then it won't free that before it begins training, is that normal?

**Comment by frosty3907** [comment]
> The "chunk" extraction method is not more than the full length split into parts. I currently train a concept LoRA of 15 shorts length 768x512 videos. Using my settings, the cache become:
> 
> ```
> 001-001_00000-037_0768x0512_hv.safetensors
> 001-001_00037-037_0768x0512_hv.safetensors
> 001-001_hv_te.safetensors
> 001-002_00000-037_0768x0512_hv.safetensors
> 001-002_00037-037_0768x0512_hv.safetensors
> 001-002_hv_te.safetensors
> 002-001_00000-037_0768x0512_hv.safetensors
> 002-001_00037-037_0768x0512_hv.safetensors
> 002-001_hv_te.safetensors
> 002-002_00000-037_0768x0512_hv.safetensors
> 002-002_hv_te.safetensors
> 003-001_00000-037_0768x0512_hv.safetensors
> 003-001_hv_te.safetensors
> 003-002_00000-037_0768x0512_hv.safetensors
> 003-002_hv_te.safetensors
> 004-001_00000-037_0768x0512_hv.safetensors
> 004-001_00037-037_0768x0512_hv.safetensors
> 004-001_hv_te.safetensors
> 004-002_00000-037_0768x0512_hv.safetensors
> 004-002_hv_te.safetensors
> 005-001_00000-037_0768x0512_hv.safetensors
> 005-001_00037-037_0768x0512_hv.safetensors
> 005-001_hv_te.safetensors
> 005-002_00000-037_0768x0512_hv.safetensors
> 005-002_00037-037_0768x0512_hv.safetensors
> 005-002_00074-037_0768x0512_hv.safetensors
> 005-002_hv_te.safetensors
> 006-001_00000-037_0768x0512_hv.safetensors
> 006-001_00037-037_0768x0512_hv.safetensors
> 006-001_hv_te.safetensors
> 007-001_00000-037_0768x0512_hv.safetensors
> 007-001_00037-037_0768x0512_hv.safetensors
> 007-001_hv_te.safetensors
> 008-001_00000-037_0768x0512_hv.safetensors
> 008-001_00037-037_0768x0512_hv.safetensors
> 008-001_00074-037_0768x0512_hv.safetensors
> 008-001_hv_te.safetensors
> 009-001_00000-037_0768x0512_hv.safetensors
> 009-001_00037-037_0768x0512_hv.safetensors
> 009-001_hv_te.safetensors
> 009-002_00000-037_0768x0512_hv.safetensors
> 009-002_00037-037_0768x0512_hv.safetensors
> 009-002_hv_te.safetensors
> ```
> 
> As you can see, the videos are splitted by 37 frames chunks and truncated to the most close part. Ofcourse there's only one text encoder cache by video.

I'm an idiot, I wasn't rebuilding the cache between changing frame extraction types.
That's why I kept getting OOM's.
Sorry.
